MaterialLibrary/src/main/java/io/doist/material/widget/FloatingActionButton.java

```diff
@@ -0,0 +1,302 @@
+            rippleDrawable.setLayerInset(1, paddingLeft, paddingTop, paddingRight, paddingBottom);
+            background = rippleDrawable;
+        } else {
+            // Lollipop androids.
+            mCircleDrawable = circleDrawable;
+            background = new InsetDrawable(
+                    new RippleDrawable(
+                            rippleColor,
+                            circleDrawable,
+                            null),
+                    paddingLeft,
+                    paddingTop,
+                    paddingRight,
+                    paddingBottom);
+        }
+
+        if (getBackground() != null) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+                setBackground(background);
+            } else {
+                setBackgroundDrawable(background);
+            }
+
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+                // Make sure the current padding is kept in pre-L androids.
+                // Otherwise, the background drawable would override the current padding.
+                setPadding(paddingLeft, paddingTop, paddingRight, paddingBottom);
+            }
+        } else {
+            throw new IllegalStateException("FloatingActionButton does not support 'android:background' attribute.");
+        }
+    }
+
+    public ColorStateList getColor() {
+        return mColor;
+    }
+
+    public void setColor(int color) {
+        setColor(ColorStateList.valueOf(color));
+    }
+
+    @SuppressLint("NewApi")
+    public void setColor(ColorStateList color) {
+        mColor = color;
+
+        if (mCircleDrawable != null) {
+            mCircleDrawable.setColor(mColor);
+        }
+
+        if (mTintDrawable != null) {
+            mTintDrawable.setTintList(mColor);
+        }
+    }
+
+    public boolean isMini() {
+        return mIsMini;
+    }
+
+    public void setIsMini(boolean isMini) {
+        if (mIsMini != isMini) {
+            setIsMiniInner(isMini);
+            requestLayout();
+            invalidate();
+        }
+    }
+
+    @SuppressLint("NewApi")
+    private void setIsMiniInner(boolean isMini) {
+        mIsMini = isMini;
+        int radiusResId = isMini ? R.dimen.fab_mini_radius : R.dimen.fab_radius;
+        mRadius = getContext().getResources().getDimensionPixelOffset(radiusResId);
+
+        if (mShadowPaint != null) {
+            setElevation(getElevation());
+        }
+    }
+
+    @SuppressLint("NewApi")
+    @Override
+    public void setElevation(float elevation) {
+        if (mInCompat) {
+            if (elevation > 0) {
+                DisplayMetrics metrics = getContext().getResources().getDisplayMetrics();
+                double lightElevation = LIGHT_ELEVATION_SP * metrics.scaledDensity;
+                double lightAngle = Math.toDegrees(Math.atan(mRadius / lightElevation));
+
+                mShadowRadius = (float) (Math.tan(Math.toRadians(lightAngle)) * (lightElevation + elevation));
```

Comment:
I think this can be simplified to `mShadowRadius = (float) (mRadius * (1 + elevation / lightElevation))`.